### ISSUE TITLE
New reference panel: Implement typography/color polish from Figma

### ISSUE THREAD

**Issue by mrnugget** [sourcegraph/sourcegraph#34201]
Implement the polish from [this Figma](https://www.figma.com/file/DZMyEJT3g2HT6HmJGNNOnW/%22Better%22-code-intel-panel?node-id=1220%3A3878):

- [x] Font colors
- [x] Element background colors, paddings, margings
- [x] Change elements and links in code view
- [x] Change wording ("100+" to "100 displayed" etc.)

Out of scope:

- Highlighting the token inside the code blob view

**Comment by jjinnii** [comment]
Providing a few items for review based on PR #34271 ! It's looking really good 😄 
- Is it possible to only have the filter bar extend to the left view of the panel? I like that you've made it sticky (which was one of your Figma questions), but I think it makes sense to only have the filter bar in the list view of the panel and not extend into the Preview. This is because the filtering affects the list view only.
<img width="874" alt="Screen Shot 2022-04-21 at 2 50 02 PM" src="https://user-images.githubusercontent.com/87138876/164541567-7bb10ba9-2667-4ba0-9d94-351678fda7ae.png">

- Is it possible to remove the horizontal scroll on the left list-view of the ref panel? When the Preview panel is exposed and/or a line of code in the list view is really long, I think it'd be cleaner to cut off code with an ellipse. (If challenging to implement, this can always be a future incremental improvement) 

- The repo name and file path in the list view use font-size-small (12px), which is distinct from the 14px base font size for the "Definitions" and "References" section headers only to help with visual hierarchy!

- The section headers seem to be very slightly off-center
<img width="1440" alt="Screen Shot 2022-04-21 at 2 39 59 PM" src="https://user-images.githubusercontent.com/87138876/164540337-3b3b8240-9696-4e40-9bf8-b0afa16f60c3.png">

- The height of the header and the filter bars are larger than my designs. For both, I maintain a height of 24 px to try and condense the content 😄 
<img width="414" alt="Screen Shot 2022-04-21 at 2 42 09 PM" src="https://user-images.githubusercontent.com/87138876/164541051-90301819-aefc-4b42-a1a3-b45e198df207.png">

- I've removed the separating line between repo, files paths, and references in my designs. At least in my Figma designs, having clean alignment is sufficient for scannability, and I think the separators add visual noise.

**Comment by mrnugget** [comment]
> * Is it possible to only have the filter bar extend to the left view of the panel? I like that you've made it sticky (which was one of your Figma questions), but I think it makes sense to only have the filter bar in the list view of the panel and not extend into the Preview. This is because the filtering affects the list view only.

Yes, that's what I meant in my PR description with "Reordering of the components so that filter/token are "inside" the left pane I will do in follow-up."

> * Is it possible to remove the horizontal scroll on the left list-view of the ref panel?

I hope so! Also noticed that. I'll see what options we have!

---

The rest of the your comments I'm ignoring for now, because it looks like you looked at the old version of the panel. All your screenshots show the version before my PR (if you look at the screenshots in my PR https://github.com/sourcegraph/sourcegraph/pull/34271 you'll see that your screenshots here are the "Before" version and not the "After" version). 

PR seems to not have been deployed to sourcegraph.com yet. If you go to sourcegraph.com now you should see the new version.

**Comment by mrnugget** [comment]
Next PR merged: https://github.com/sourcegraph/sourcegraph/pull/34342 This one should go live in the next few hours (hopefully). But the PR description contains screenshots.

I'll keep this ticket open for any design feedback on latest iteration.

**Comment by mrnugget** [comment]
Just to be explicit: latest changes are live.

**Comment by jjinnii** [comment]
Thanks @mrnugget  It looks great! I think the only feedback I want to resolve before we share to #feedback-dogfood is that it's hard to distinguish between the different file paths. I think a couple things might resolve this: 

- In my designs, I have dividers between repository results, and I included a [quick view here](https://www.figma.com/file/DZMyEJT3g2HT6HmJGNNOnW/%22Better%22-code-intel-panel?node-id=1326%3A3875) where I've also added dividers between file result groupings. 
- I realize that the .75 rem for the repo and file header is smaller seems really small. I changed these headers to .875 rem to match the code-font for the results, and I think it reads a lot better. Thoughts?
<img width="1438" alt="Screen Shot 2022-04-26 at 4 47 18 PM" src="https://user-images.githubusercontent.com/87138876/165398142-45b841c6-8fda-4945-b257-5528b1b992a7.png">
